executor_manager/routers/routers.py

```
@@ -1332,5 +1332,175 @@ async def task_heartbeat(task_id: str, http_request: Request):
+        logger.info(
+            f"[Archive] Successfully archived task {request.task_id}, "
+            f"size={archive_result.get('size_bytes')} bytes"
+        )
+
+        return archive_result
+
+    except HTTPException:
+        raise
+    except httpx.HTTPError as e:
+        logger.error(f"[Archive] HTTP error for task {request.task_id}: {e}")
+        raise HTTPException(status_code=500, detail=f"HTTP error: {str(e)}")
```

Ruff (0.15.7)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)